packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -294,23 +325,69 @@ const ProductDetail = () => {
         })
     }
 
-    const handleAddToCart = async (productSelectionValues) => {
-        try {
-            const productItems = productSelectionValues.map(({variant, quantity}) => ({
-                productId: variant.productId,
-                price: variant.price,
-                quantity
-            }))
+    const [pickupInStoreMap, setPickupInStoreMap] = useState({})
 
-            await addItemToNewOrExistingBasket(productItems)
+    const handlePickupInStoreChange = (productId, checked) => {
+        setPickupInStoreMap((prev) => ({
+            ...prev,
+            [productId]: checked
+        }))
+    }
 
-            einstein.sendAddToCart(productItems)
+    const addToCartModal = useAddToCartModalContext()
 
-            // If the items were successfully added, set the return value to be used
-            // by the add to cart modal.
+    const handleAddToCart = async (productSelectionValues = []) => {
+        try {
+            const productItems = productSelectionValues.map((item) => {
+                const {variant, quantity} = item
+                // Use variant if present, otherwise use the main product
+                const prod = variant || item.product || product
+                const prodKey = prod.productId || prod.id
+                let result = {
+                    productId: prod.productId || prod.id, // productId for variant, id for product
+                    price: prod.price,
+                    quantity
+                }
+                // Robustly fetch inventoryId from localStorage if pickupInStore is true
+                if (pickupInStoreMap[prodKey]) {
+                    const siteId = site?.id || (window.SFCC && window.SFCC.siteId)
+                    const storeInfoKey = `store_${siteId}`
```

Comment:
Not sure if I understand this variable here. All the changes seem to get this value from the storage, but I could not find a place where it is being set into the storage

Reply:
It is set in the file: packages/template-retail-react-app/app/components/store-locator-modal/stores-list.jsx
const StoresList